lib/scope.js

```diff
@@ -413,6 +429,147 @@ module.exports = {
     return winner;
   },  // Ends scope.tapElement()
 
+  // Handle server reload errors
+  server_statuses: [],
+  server_check_interval_ms: 1 * 1000,
+  stop_tracking_server_status: false,
+  track_server_status: async function( scope ) {
+    /** Updates server_statuses list by checking if server is responding until,
+    *    eventually, it is told to stop. */
+
+    while ( !scope.stop_tracking_server_status ) {
+      // This code assumes a reasonable response time would be 2 seconds if the
+      // server is up and that a server reload/restart won't take less than 2 seconds.
+      // Also, it's non-blocking, so it doesn't slow down tests
+      let is_up = await server.is_server_responding({ timeout: scope.server_check_interval_ms });
```

Comment:
I think this assumes that `is_server_responding` will always take `timeout` ms to respond, but it doesn't have to take `timeout` seconds. When it returns sooner, we just happily check if it's responding at a faster rate, which I don't think is what we want.

Not sure the javascripty way of doing things, but I would likely get the current time before we call the function, get how long did pass after we call it (current time again - before time), and then sleep for that long.

Reply:
Yep, that looks good to me!

Reply:
Is this ready for approval?

Reply:
sorry, yeah